src/backend/OpenCL/CHIPBackendOpenCL.cc

```diff
@@ -1579,36 +1579,53 @@ void CHIPBackendOpenCL::initializeImpl(std::string CHIPPlatformStr,
 
   StrStream.str("");
 
-  StrStream << "OpenCL Devices of type " << CHIPDeviceTypeStr
-            << " with SPIR-V_1 support:\n";
-  std::vector<cl::Device> SpirvDevices;
+  StrStream << "OpenCL Devices of type " << CHIPDeviceTypeStr << ": "
+            << std::endl;
+  std::vector<cl::Device> SupportedDevices;
   std::vector<cl::Device> Dev;
   Err = SelectedPlatform.getDevices(SelectedDevType, &Dev);
   CHIPERR_CHECK_LOG_AND_THROW(Err, CL_SUCCESS, hipErrorInitializationError);
   for (auto D : Dev) {
-    std::string Ver = D.getInfo<CL_DEVICE_IL_VERSION>(&Err);
-    if ((Err == CL_SUCCESS) && (Ver.rfind("SPIR-V_1.", 0) == 0)) {
-      std::string DeviceName = D.getInfo<CL_DEVICE_NAME>();
-      StrStream << DeviceName << "\n";
-      SpirvDevices.push_back(D);
+
+    std::string DeviceName = D.getInfo<CL_DEVICE_NAME>();
+
+    StrStream << DeviceName << " ";
+    std::string SPIRVVer = D.getInfo<CL_DEVICE_IL_VERSION>(&Err);
+    if ((Err != CL_SUCCESS) ||
+        (SPIRVVer.rfind("SPIR-V_1.", 0) == std::string::npos)) {
+      StrStream << " no SPIR-V support." << std::endl;
```

Comment:
Should we avoid [std::endl](https://llvm.org/docs/CodingStandards.html#avoid-std-endl) even tough we are not dealing with an IO-stream for style consistency?

Reply:
I've understood a benefit (in addition to the flush) you get from std::endl is that it can also add a DOS/Windows line feed if running on Windows.

Reply:
https://stackoverflow.com/questions/1651936/changing-stdendl-to-put-out-crlf-instead-of-lf ok, \n char seems to get expanded to Windows linefeeds also. It shows that how much I've programmed on Windows ;) Anyhow, I don't think it matters if it flushes that stream here.

Reply:
OK. It was new info to me that the implementation is supposed to check the outputted chars and replace \n with \r\n for Windows.